teps/0086-changing-the-way-result-parameters-are-stored.md

```diff
@@ -115,6 +120,58 @@ Additionally, this will help projects that wrap/abstract Tekton where users unde
+controller will access the stdout logs of the `Sidecar`, extract the `Result` and its contents during reconciliation.
+After the `Steps` have terminated, the `TaskRun` controller will attach the `Results` from the logs of the `Sidecar` 
+instead of using the termination message (hence bypassing the 4K limit) to update the `Results` fields in the CRD. 
+This method keeps the rest of the current functionality identical and does not require any external storage mechanism.
+For further details, see the [demonstration][demo] of the proof of concept.
+
+#### Feature Gates for Sidecar Logs
+
+This solution will be gated using a `enable-sidecar-logs-results` feature flag which will default to `"false"` and
+users can set it to `"true"` to enable it. This provides an opportunity to experiment with this solution to provide
+`Results` within the CRDs as we figure out the solutions for external storage.
+
+In [TEP-0100][tep-0100] we proposed changes to `PipelineRun` status to reduce the amount of information stored about
+the status of `TaskRuns` and `Runs` to improve performance, reduce memory bloat and improve extensibility. Now that
+those changes have been implemented, the `PipelineRun` status is set up to handle larger `Results` without 
+exacerbating the performance and storage issues that were there before. For `ChildReferences` to be populated, the
+`embedded-status` must be set to `"minimal"`. Thus, will require that minimal embedded status is enabled during the
+migration until it becomes the only supported status. This requirement also makes the behavior clearer to users -
+auto-adding the minimal status when users have not enabled it makes the user experience opaque and surprising.
+
+#### Design Details of Sidecar Logs
+
+In order to parse the `Results` volume and output the contents to stdout in the `Container`, the `Sidecar` will run 
+a script that:
+- is auto-generated based on the required `Results` - identified from `taskSpec.results` field.
+- periodically checks for new files in the directory `/tekton/results`.
+- When a new `Result` is found, it prints it to stdout in a parsable pattern.
+- When all the expected `Results` are found, it breaks out of the periodic loop and exits.
+
+#### Caveats of Sidecar Logs
+
+- If a `Result` file is created first and then contents are written to it, and the `Result` loop happens to read the 
+file in the interim then only partial contents will be considered as `Results`.
+- If a `Step` fails to produce a `Result`, the `Sidecar` will continue to look for the `Result` until it times out.
+- No guarantees on when logs will be available - would have to wait for this before considering a `TaskRun` complete.
+- No guarantee you'll be able to access a log before it disappears e.g. logs will not be available via the k8s API
+  once a `Pod` is deleted.
+- The storage of the result parameter may still be limited by the [1.5 MB CRD size][crd-size]. 
```

Comment:
I wouldn't document saying the results can be 1.5MB. Complex tasks with multiple steps (which is generally the case) might take away sufficient storage space. We can document the limitation saying the maximum allowed result size depends on the CRD, the max allowed CRD size - the actual CRD size. Also, this is split among all the results of a task.

How is the storage size distributed if my task has multiple results?

Reply:
I agree with your statement. There is no firm upper limit to the results. As long as the total CRD limit is not crossed, its fair game. If we have 100s of results, that we store in the CRD then they have to share the CRD limit.

Reply:
that makes it much clearer, updated this caveat